offload/plugins-nextgen/amdgpu/CMakeLists.txt

```diff
@@ -28,7 +28,11 @@ if(NOT (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86_64)|(ppc64le)|(aarch64)$" AND CMAKE
 endif()
 
 # Create the library and add the default arguments.
-add_target_library(omptarget.rtl.amdgpu AMDGPU)
+if("AMDGPU" IN_LIST LLVM_TARGETS_TO_BUILD)
```

Comment:
Why don't we put this logic into `add_target_library`?

Reply:
I'd make this same as the argument `LINK_COMPONENTS` of `add_llvm_library`, since it is not always required.  Besides, given `add_target_library` is only for plugin, and only "host" plugin doesn't require that, we can assume almost all plugins need the corresponding target. `add_target_library(omptarget.rtl.amdgpu AMDGPU "")` isn't really beautiful. We can even force `lib_name` to be the target name.

Reply:
Honestly we should probably rename the `amdgpu` plugin to `hsa` anyway? Since the important distinction is the runtime, not the target architecture. Potentially in the future we'll have things like OpenCL or Vulkan I guess? Which are supposed to be generic in their own right.
So, what we could do is just `target_library(libomptarget.amdgpu AMDGPU)` or something, but I think that might require something else.